samples/net/rpl_border_router/sample.yaml

```diff
@@ -0,0 +1,8 @@
+sample:
+    name: RPL Border Router
+tests:
+-   test:
+        build_only: true
+        platform_whitelist: frdm_k64f qemu_x86
```

Comment:
if this sample is not intended for qemu, why are we whitelisting it here? why do we have a configuration for quark se but we are not building it? Also, the frdm board does not have 15.4.. I suggest removing the whitelist and adding:

depends_on: ieee802154

and make this more generic, we are going back to adding board specific configurations, something we need to stop doing.